### PR TITLE
Token selection modal

### PR DESCRIPTION
**Pull request by AtelyPham**
- style: improve mobile display of bridge screen
- feat: implement chain selection modal
- chore: fix modal height and remove unused code
- chore: remove margin for consistent spacing for mixer button between srcreens
- style: improve responsive on amount session
- refactor: `useBreakpoint` hook logic and change to export named function
- chore: import packages instead relative import
- feat: add token selection modal
- style: fix style for scrollable

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

CleanShot 2022-06-11 at 02 02 50

Issue Number: https://github.com/webb-tools/webb-dapp/issues/396

## What is the new behavior?

CleanShot 2022-06-11 at 02 03 53

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

## Other information
Checkout from from https://github.com/webb-tools/webb-dapp/pull/418 to reuse the features, you should review that https://github.com/webb-tools/webb-dapp/pull/418 first.

### CODE REVIEW

**Review by drewstone — APPROVED** [review]
LGTM!

**Comment by dutterbutter** [comment]
@AtelyPham great job on this! There seems to be a small issue when running with protocol-substrate. When I select a token, I see two options. (1) for webb tokens, (2) for test tokens, however when I select test tokens the field is never populated with that token. Seems to work fine in the mixer tab but not in bridge tabs. 


https://user-images.githubusercontent.com/29983536/173147586-994ac14b-a9e2-402c-ae17-bc87d5e196c3.mov

 Let me know if you can reproduce @AtelyPham.

**Review by dutterbutter — CHANGES_REQUESTED** [review]
Please see issue while running with protocol-substrate.

**Comment by AtelyPham** [comment]
I'm not sure whether I'm correct or not for checking the supported currency in the commit https://github.com/webb-tools/webb-dapp/pull/423/commits/a8f7894168a9e9fa68414287becfe70b9a9fa121. Please help me check again, thanks 😊.

File: packages/ui-components/src/Inputs/ChainSelection/ChainSelection.tsx [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+                </Typography>
+              </ListItemText>
+              {selected && <CircleStatus />}
+            </ListItem>
+          );
+        })}
+      </>
+    ),
+    [displayChainTypeIds, selectedChain, getChainFromChainTypeId, onClose, onChange]
+  );
```
[the remark this answers is not in the document]
Comment (AtelyPham) [reply packages/ui-components/src/Inputs/ChainSelection/ChainSelection.tsx:196]: I refactored in the commit https://github.com/webb-tools/webb-dapp/pull/418/commits/e175b21c98fd555d826d0e43e891cea422fe13cd. Please review it when you have time, then I will merge it into this branch.

**Review by drewstone — CHANGES_REQUESTED** [review]
Merge conflicts need resolving here.

**Review by AhmedKorim — APPROVED** [review]

**Pull request merged**